pkg/resource/logging-credentials/logging_operator_secrets.go

```diff
@@ -0,0 +1,75 @@
+package loggingcredentials
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LoggingCredentialsSecretMeta returns metadata for the logging-operator credentials secret.
+func LoggingCredentialsSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      "logging-credentials",
+		Namespace: "monitoring",
+	}
+}
+
+// Generate a random 20-characters password
+func genPassword() string {
+	const length = 20
+
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"~=+%^*/()[]{}/!@#$?|")
+
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		b.WriteRune(chars[rand.Intn(len(chars))])
+
+	}
+	str := b.String() // E.g. "ExcbsVQs"
+	return str
+}
+
+// GenerateObservabilityBundleConfigMap returns a configmap for
+// the observabilitybundle application to enable promtail.
+func GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) (v1.Secret, error) {
```

Comment:
The function returns a secret and an error. But I think most of the times where we need the secret, we actually need a pointer to a secret, so it may be more convenient to actually return it from here

```
GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) (*v1.Secret, error)
```

Thinking about this function, if we are never going to return an error, we don't need the second returned value and we could leave the signature as
```
GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) *v1.Secret
```

Reply:
Makes sense. Especially since I've already been tricked by using secret instead of pointer, and didn't understand why my original resource was updated! :sweat_smile:

Reply:
Done in commit 81c1d49d6228b4aed1d0f5028ebe91129e6ed267